sonar-plugin-api/src/main/java/org/sonar/api/server/authentication/UserIdentity.java

```diff
@@ -123,6 +149,21 @@ public Builder setEmail(@Nullable String email) {
       return this;
     }
 
+    /**
+     * Set groups of the user.
+     *
+     * If list of groups is null, no synchronization will be done.
+     *
+     * @since 5.5
+     */
+    public Builder setGroups(@Nullable Set<String> groups) {
+      if (groups != null) {
```

Comment:
Missing validations of format of group name. What's the max size ? What are the accepted characters (slash, whitespace, ...) ?